### ISSUE TITLE
Enable Azure Auto Heal for Umbraco Cloud hosted sites

### ISSUE THREAD

**Issue by lucas-whiteley** [umbraco/Umbraco.Cloud.Issues#843]
### Issue description

When sites go down in Umbraco cloud, they are not auto restarted. This could be fixed by enabling Azure Auto Heal which will restart problematic sites. This would reduce downtime for websites on the platform.

**Comment by c9mb** [comment]
Yes - I don't know if it's related, but when the worker-role is moved between hosts - which has been happening a lot recently - the sites also appear to be left in a shut-down state, and wait for the first request to initiate a cold-start. Because they have been moved, this means there is some compile action required, and this can be quite a lengthy delay for the unfortunate soul who lucks out on that - and  leads to a bit of aggro with clients.

**Comment by BenDuguid-MRM** [comment]
Have you tried using the [Application Initialization module](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/advanced?view=aspnetcore-8.0#application-initialization-module)?

The IIS Application Initialization module is installed in Azure for app services, but you need to use the middleware and configure your web app to use it:

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <location path="." inheritInChildApplications="false">
    <system.webServer>
      <applicationInitialization doAppInitAfterRestart="true" />
    </system.webServer>
  </location>
</configuration>
```

**Comment by jemayn** [comment]
Hey everyone!

Autoheal has always been on since it was enabled by Azure, if you are seeing issues with sites not being started I would recommend you reach out to Cloud support.

There is a separate feature request for opting out of it you can follow if that is something you need: https://github.com/umbraco/Umbraco.Cloud.Issues/discussions/1007
